src/persist-client/src/internal/machine.rs

```diff
@@ -871,6 +922,18 @@ where
                     }
                 };
 
+                // Sanity check that all state transitions have special case for
+                // being a tombstone. The ones that do will return a Break and
+                // return out of this method above. The one exception is adding
+                // a rollup, because we want to be able to add a rollup for the
+                // tombstone state.
+                if was_tombstone_before && !is_rollup {
+                    panic!(
```

Comment:
if we have a tombstoned shard presumably no one can read from it anymore, so I'm not sure we want the crash the whole process if we hit this case. maybe this can be a debug assert?